src/kaleidoscope/hardware/keyboardio/model01/Model01KeyScanner.cpp

```diff
@@ -0,0 +1,191 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-Hardware-Model01 -- Keyboard.io Model01 hardware support for Kaleidoscope
+ * Copyright (C) 2017-2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef ARDUINO_AVR_MODEL01
+#ifndef KALEIDOSCOPE_VIRTUAL_BUILD
+
+#include <Kaleidoscope.h>
+#include <KeyboardioHID.h>
+#include <avr/wdt.h>
+
+namespace kaleidoscope {
+namespace hardware {
+namespace keyboardio {
+
+KeyboardioScanner Model01KeyScanner::leftHand(0);
+KeyboardioScanner Model01KeyScanner::rightHand(3);
+
+keydata_t Model01KeyScanner::leftHandState;
+keydata_t Model01KeyScanner::rightHandState;
+keydata_t Model01KeyScanner::previousLeftHandState;
+keydata_t Model01KeyScanner::previousRightHandState;
+keydata_t Model01KeyScanner::leftHandMask;
+keydata_t Model01KeyScanner::rightHandMask;
+
+void Model01KeyScanner::enableScannerPower(void) {
```

Comment:
technically, this is for a lot more than the keyscanners. And doesn't actually affect the left-side keyscanner.

Reply:
This was the historic name of the function, and `ScannerPower` suggested it's related to the key scanner. Where should this be moved, and under what name?

Reply:
I will 100% agree that the old naming sucked.  it's really "led power, and right-hand power" which is not a great name or all that easy to express :)

Reply:
I think I found a nice way to express that: `Model01Hands::setup()`!